Clean up workflow actions

seallard opened the pull request:
This PR cleans up the github actions in this repo. Separate actions for linting, testing, pushing to dockerhub and bumping the version have been created. 

New behavior: the version is bumped and images are pushed automatically to dockerhub when a PR is merged to master. The images will be tagged like `2.1.6` etc. Also replaced the existing linting with `black` since that is what we use in other repos. Let me know if you prefer the flake/mypy setup. If this fork is ever to be merged to the source (?) it is probably better to keep the existing linter.

I don't have the access rights in this repo to do so, but the default merge strategy should be set to "squash and merge". It might also be necessary to adjust some settings for the bump action to be triggered.

Would also be nice to block merging until a review has been made, tests and linting passes.

### Fixed
- Remove `ci.yml` action (unused)
- Remove `docker-image-manual.yml ` action (replaced)
- Remove `docker-image.yml` action (replaced)
- Remove `pr.yml`(replaced)
- Remove `release.yml` (replaced)
- Add action automatically bumping version
- Add action automatically pushing images to dockerhub
- Add action running integration tests
- Add action for linting

rannick commented:
Great work!
Did the setup changes you suggested (+added bioinfo as admin), corrected the integration test to what we are currently using and switched to black.

seallard commented:
> Great work!
> Did the setup changes you suggested (+added bioinfo as admin), corrected the integration test to what we are currently using and switched to black.

Great! Noticed I missed a config file to ensure the bump works. I'll patch it tomorrow
